src/objectExplorer/objectExplorerService.ts

```diff
@@ -212,6 +212,15 @@ export class ObjectExplorerService {
         }
     }
 
+    /**
+     * Helper to show the Add Connection node
+     */
+    private getAddConnectionNode(): AddConnectionTreeNode[] {
+        this._rootTreeNodeArray = [];
+        this._objectExplorerProvider.objectExplorerExists = true;
```

Comment:
Few non-blocking things :

- Why is this separated out? It's only called here so having a separate function for something this simple doesn't seem super useful
- The function name doesn't really describe what's happening - in particular setting oeExists = true and clearing the cached node array
- Why return this as an array? Seems clearer to just return the single node

Reply:
It's separated out because it's being used twice. Don't want reduntant code in the `getChildren` function. Will change the name.